lua/entities/starfall_processor/cl_init.lua

```diff
@@ -140,7 +140,7 @@ net.Receive("starfall_processor_used", function(len)
 
 	instance:runScriptHook("starfallused", instance.WrapObject( activator ), instance.WrapObject( used ))
 
-	if activator == LocalPlayer() and instance.player ~= SF.Superuser and instance.permissionRequest and instance.permissionRequest.showOnUse and not SF.Permissions.permissionRequestSatisfied( instance ) and not IsValid(chip.permPanel) then
+	if activator == LocalPlayer() and instance.player ~= SF.Superuser and instance.permissionRequest and instance.permissionRequest.showOnUse and not SF.Permissions.permissionRequestSatisfied( instance ) and not IsValid(chip.permPanel) and not SF.showingPermissionPrompt then
```

Comment:
I already added the IsValid(chip.permPanel) to fix spamming

Reply:
I was still able to get multiple prompts with 2 different chips.

Reply:
I think I'll replace `chip.permPanel` with `SF.permPanel` and use that in place of `SF.showingPermissionPrompt`. This would still prevent that unlikely scenario while making the code as clean as it was before.

Reply:
Ok, that works.